packages/@uppy/aws-s3/src/MiniXHRUpload.js

```diff
@@ -10,22 +10,12 @@ const { internalRateLimitedQueue } = require('@uppy/utils/lib/RateLimitedQueue')
 
 // See XHRUpload
 function buildResponseError (xhr, error) {
-  // No error message
-  if (!error) error = new Error('Upload error')
-  // Got an error message string
-  if (typeof error === 'string') error = new Error(error)
-  // Got something else
-  if (!(error instanceof Error)) {
-    error = Object.assign(new Error('Upload error'), { data: error })
-  }
-
-  if (isNetworkError(xhr)) {
-    error = new NetworkError(error, xhr)
-    return error
-  }
+  if (isNetworkError(xhr)) return new NetworkError(error, xhr)
 
-  error.request = xhr
-  return error
+  const err = new Error('Upload error')
+  err.cause = error
+  err.request = xhr
+  return err
```

Comment:
From the proposal link you shared it seems this is also possible

```suggestion
return new Error('Upload error', { cause: error, request: xhr })
```

Reply:
```suggestion
  // TODO: when we drop support for browsers that do not support this syntax, use:
  // return new Error('Upload error', { cause: error, request: xhr })
  const err = new Error('Upload error')
  err.cause = error
  err.request = xhr
  return err
```